sample/redo/lib/main.dart

```diff
@@ -183,50 +183,32 @@ class _MyHomePageState extends State<MyHomePage> {
     // await _init();
   }
 
-  Mutator<void, Map<String, dynamic>> _todoCreateMutator;
-  Mutator<void, Map<String, dynamic>> _todoDoneMutator;
-  Mutator<void, Map<String, dynamic>> _todoRemoveMutator;
-  Mutator<void, Map<String, dynamic>> _todoReorderMutator;
+  Mutator<void, Map<String, dynamic>> _createTodo;
+  Mutator<void, Map<String, dynamic>> _removeTodo;
+  Mutator<void, Map<String, dynamic>> _updateTodo;
 
   _registerMutations() {
-    _todoCreateMutator = _replicache.register('todo-create',
+    _createTodo = _replicache.register('createTodo',
         (tx, Map<String, dynamic> args) async {
-      int id = args['id'];
-      String text = args['text'];
-      int listId = args['listId'];
-      Iterable<Todo> todos = await todosInListFromTx(tx, listId);
-      final order = newOrderBetween(todos.isEmpty ? null : todos.last, null);
-      await _write(tx, Todo(id, listId, text, false, order));
+      await _write(tx, Todo.fromJson(args));
     });
 
-    _todoDoneMutator = _replicache.register('todo-done',
-        (tx, Map<String, dynamic> args) async {
-      int id = args['id'];
-      bool complete = args['complete'];
-      final todo = await _read(tx, id);
-      if (todo == null) {
-        return;
-      }
-      todo.complete = complete;
-      await _write(tx, todo);
-    });
-
-    _todoRemoveMutator = _replicache.register('todo-remove', (tx, args) async {
+    _removeTodo = _replicache.register('removeTodo', (tx, args) async {
       int id = args['id'];
       await _del(tx, id);
     });
 
-    _todoReorderMutator =
-        _replicache.register('todo-reorder', (tx, args) async {
+    _updateTodo = _replicache.register('updateTodo', (tx, args) async {
       int id = args['id'];
-      double order = args['order'];
       final todo = await _read(tx, id);
       if (todo == null) {
         print('Warning: Possible conflict - Specified Todo $id is not present.'
             ' Skipping reorder.');
         return;
       }
-      todo.order = order;
+      todo.text ??= args['text'] ?? todo.text;
+      todo.complete = args['complete'] ?? todo.complete;
```

Comment:
why are the operators different on the different lines?

Reply:
bug